workflows/chrcheck/main.nf

```diff
@@ -0,0 +1,79 @@
+include { VCF_CHR_EXTRACT         } from '../../modules/local/vcf_chr_extract'
+include { BAM_CHR_EXTRACT         } from '../../modules/local/bam_chr_extract'
+include { BAM_CHR_RENAME_SAMTOOLS } from '../../subworkflows/local/bam_chr_rename_samtools'
+include { VCF_CHR_RENAME_BCFTOOLS } from '../../subworkflows/local/vcf_chr_rename_bcftools'
+include { checkChr                } from '../../subworkflows/local/utils_nfcore_chrcheck_pipeline'
+include { diffChr                 } from '../../subworkflows/local/utils_nfcore_chrcheck_pipeline'
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    RUN MAIN WORKFLOW
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+workflow CHRCHECK {
+    take:
+        ch_input // [[id], file, index, [chr]]
+
+    main:
+        ch_versions = Channel.empty()
+        // Split the input between VCF and BAM files
+        ch_input = ch_input.branch{
+            bam: it[1] =~ 'bam|cram'
+            vcf: it[1] =~ 'vcf|bcf'
+            other: it[1].size() > 0
+            empty: true
+        }
+
+        ch_input.other.map {
+            error "File: ${it[1]} is not a VCF, BCFT or BAM, CRAM file."
+        }
+
+        // Check if channel is empty
+        chr_vcf_disjoint = Channel.empty()
+        // Extract the contig names from the VCF files
+        VCF_CHR_EXTRACT(ch_input.vcf.map{ meta, file, _index, _chr -> [meta, file] })
+        ch_versions = ch_versions.mix(VCF_CHR_EXTRACT.out.versions)
+        chr_vcf_disjoint = checkChr(VCF_CHR_EXTRACT.out.chr, ch_input.vcf)
```

Comment:
What does disjoint mean here?

Reply:
It is all the vcf files that do not have the same contigs names.

Reply:
I've changed `chr_vcf_disjoint` to `ch_vcf_branched`.
Is it better ?

Reply:
What about `ch_vcf_split`? (asked seqera AI for a better name 😃 )

Reply:
That would be better 😄

Reply:
Done

Reply:
Then the name would need to be updated on your other PR too.